src/misc/rwqueue.cpp

```diff
@@ -244,24 +264,34 @@ std::optional<T> RWQueue<T>::Dequeue()
 }
 
 template <typename T>
-bool RWQueue<T>::BulkDequeue(std::vector<T>& into_target, const size_t num_requested)
+size_t RWQueue<T>::BulkDequeue(std::vector<T>& into_target, const size_t num_requested)
 {
-	constexpr size_t min_items = 1;
-	assert(num_requested >= min_items);
-
-	if (into_target.size() != num_requested) {
+	if (into_target.size() < num_requested) {
 		into_target.resize(num_requested);
 	}
 
-	auto target_start  = into_target.begin();
+	const auto num_dequeued = BulkDequeue(into_target.data(), num_requested);
+
+	// cap off the target vector to match the number that were dequeued
+	into_target.resize(num_dequeued);
+
+	return num_dequeued;
+}
+
+template <typename T>
+size_t RWQueue<T>::BulkDequeue(T* const into_target, const size_t num_requested)
+{
+	assert(into_target);
+	auto target_start  = into_target;
 	auto num_remaining = num_requested;
 
 	while (num_remaining > 0) {
 		std::unique_lock<std::mutex> lock(mutex);
 
-		const auto num_items = std::max(min_items,
-		                                std::min(num_remaining,
-		                                         queue.size()));
+		constexpr size_t MinItems = 1;
+		const auto num_items      = std::max(MinItems,
+                                                std::min(num_remaining,
+                                                         queue.size()));
```

Comment:
Hmm, I thought you changed this to use `clamp` from `math_utils`?

Reply:
Ok... I was reviewing commit by commit. Found the change in a later commit 😄

Reply:
Had me worried I lost some commits or something! Glad it's addressed!